pkg/issuer/vault/issue.go

```diff
@@ -163,9 +166,52 @@ func (v *Vault) initVaultClient() (*vault.Client, error) {
 		return client, nil
 	}
 
+	kubernetesAuth := v.issuer.GetSpec().Vault.Auth.Kubernetes
+	if kubernetesAuth.Role != "" {
+		token, err := v.requestTokenWithKubernetesAuth(client, kubernetesAuth.Role)
+		if err != nil {
+			return nil, fmt.Errorf("error reading Kubernetes service account token from %s: %s", kubernetesServiceTokenPath, err.Error())
+		}
+		client.SetToken(token)
+ 		return client, nil
+	}
+
 	return nil, fmt.Errorf("error initializing Vault client. tokenSecretRef or appRoleSecretRef not set")
 }
 
+func (v *Vault) requestTokenWithKubernetesAuth(client *vault.Client, role string) (string, error) {
+	// read jwt from container
+	jwt, err := ioutil.ReadFile(kubernetesServiceTokenPath)
+	if err != nil {
+		return "", fmt.Errorf("error reading Kubernetes service account token from %s: %s", kubernetesServiceTokenPath, err.Error())
+	}
+ 	parameters := map[string]string{
+		"role": role,
+		"jwt":  string(jwt),
+	}
+ 	url := "/v1/auth/kubernetes/login"
```

Comment:
This path needs to be configurable. Perhaps:

```yaml
auth:
  kubernetes:
    mountPath: auth/kubernetes
    serviceAccountSecretRef:
      name: svc-acct-for-auth
```

Reply:
> @kunickiaj
> 
> I didn't experiment with this Vault auth method much yet but maybe you could use the helper method in test/e2e/framework/util.go to fetch the config:
> 
> ```go
> kubeConfig, err := LoadConfig(TestContext.KubeConfig, TestContext.KubeContext)
> if (err != nil) {
> ...
> }
> 
> apiHost := kubeConfig.Host
> caCert := kubeConfig.TLSClientConfig.CAFile
> ```
> 
> Would it give you what you need to setup the Kubernetes auth backend in Vault afterwards?

What's the TestContext here? Was looking at augmenting VaultInitializer to perform the setup but not sure where to pass down the kubeconfig/context from.